homeassistant/components/trafikverket_train/sensor.py

```diff
@@ -0,0 +1,185 @@
+"""Train information for departures and delays, provided by Trafikverket."""
+
+from datetime import date, datetime, timedelta
+import logging
+
+from pytrafikverket import TrafikverketTrain
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import CONF_API_KEY, CONF_NAME, CONF_WEEKDAY, WEEKDAYS
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_TRAINS = "trains"
+CONF_FROM = "from"
+CONF_TO = "to"
+CONF_TIME = "time"
+
+ATTR_CANCELED = "canceled"
+ATTR_DELAY_TIME = "number_of_minutes_delayed"
+ATTR_PLANNED_TIME = "planned_time"
+ATTR_ESTIMATED_TIME = "estimated_time"
+ATTR_ACTUAL_TIME = "actual_time"
+ATTR_OTHER_INFORMATION = "other_information"
+ATTR_DEVIATIONS = "deviations"
+
+ICON = "mdi:train"
+SCAN_INTERVAL = timedelta(minutes=5)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_API_KEY): cv.string,
+    vol.Required(CONF_TRAINS): [{
+        vol.Required(CONF_NAME): cv.string,
+        vol.Required(CONF_TO): cv.string,
+        vol.Required(CONF_FROM): cv.string,
+        vol.Optional(CONF_TIME): cv.time,
+        vol.Optional(CONF_WEEKDAY, default=WEEKDAYS):
+            vol.All(cv.ensure_list, [vol.In(WEEKDAYS)])}]
+})
+
+
+async def async_setup_platform(
+        hass, config, async_add_entities, discovery_info=None):
+    """Set up the departure sensor."""
+    httpsession = async_get_clientsession(hass)
+    train_api = TrafikverketTrain(httpsession, config.get(CONF_API_KEY))
+    sensors = []
+    station_cache = {}
+    for train in config.get(CONF_TRAINS):
```

Comment:
Use `dict[key]` for required config keys and keys with default config schema values.

Reply:
Yep, i will fix. Should i use `dict[key]` for all occations where i have `dict.get[key]` or are there any cases where `dict.get` should be preferred?

Reply:
Only use `dict.get` if we don't know if the key will be present in the dict or not.

Reply:
Pushed, could not change for `train.get(CONF_TIME)` since `train[CONF_TIME]` would translate to `train['time']`

```
line 78, in async_setup_platform
    train[CONF_TIME])
KeyError: 'time'
```

Reply:
Yeah, `CONF_TIME` is an optional config item.